auslib/test/web/test_client.py

```diff
@@ -805,6 +805,24 @@ def testAvastURLsWithGoodQueryArgs(self):
 </updates>
 """)
 
+    def testForceParamWithBadInputs(self):
+        force_output = """<?xml version="1.0"?>
+<updates>
+    <update type="minor" version="1.0" extensionVersion="1.0" buildID="2">
+        <patch type="complete" URL="http://a.com/z" hashFunction="sha512" hashValue="4" size="3"/>
+    </update>
+</updates>
+"""
+        inputs = [
+            '',
+            '1" name="Firefox 54.0" isOSUpdate="false" installDate="1498012260998',
+            '1)',
+            '"|sleep 7 #'
+        ]
+        for i in inputs:
```

Comment:
You can use hypothesis package to help you on test cases like that.
Here an usage example:
https://github.com/mozilla/balrog/blob/master/auslib/test/web/test_client.py#L905

Take a look at Hypothesis docs too if you want:
https://hypothesis.readthedocs.io/en/latest/#

Reply:
That looks interesting. Will use it and refactor the test cases.